Add an "ExcludeFromManifest" option to manifest generation

chcosta opened the pull request:
Fixes https://github.com/dotnet/arcade/issues/1842

Unblocks core-sdk publishing.  Validated in https://dev.azure.com/dnceng/internal/_build/results?buildId=78284

Allows us to do https://github.com/chcosta/core-sdk/blob/publish-to-bar/eng/Publishing.props#L35, to prevent a file from being included in a manifest.

@jcagme @JohnTortugo , I updated `PushToAzureDevOpsArtifacts.cs` in this change for consistency, but that code will need to be refactored when that file does more than just manifest generation because the goal is to allow the items to be published, but not included in the manifest (as-is, they would be excluded from both).

jcagme commented:
Change looks good and I actually like the idea of supporting exclusion. 

About not publishing SVGs tho, before the holiday break I talked to @mmitche about whether or not we should push all the built assets to BAR and we agreed that we should. No repo will take a dependency on them but it could be useful metadata. Also, I remember fixing the version issue for SVGs so I'll need to look at what is wrong.

I'm ok with not including them in BAR we just should be all in the same page.

chcosta commented:
Thanks @jcagme ,  I'll merge this change as soon as CI is green (it appears to be failing for unrelated reasons).  @mmitche , let's talk about this further.  Personally, I think we should move forward with this solution and if we see a need for the svg files in the future (or think this is going to set a bad precedent) then I can look at addressing those concerns but at least core-sdk publishing won't be blocked.

jcagme commented:
Yeah, will do when I have find the bug
